### PR TITLE
Update MCNP material reader to support default library identifier

### LINKED ISSUE #1165: Allow specification of default MCNP table ids for materials
The MCNP material writer currently supports metadata "table_ids" to specify a particular cross-section table for any set of nuclides desired by the user.

In some cases it may also be useful to specify a default cross-section set for all nuclides using the syntax "nlib=81c" or "hlib=70h".

This might be possible with metadata "default_table_ids" that is a dictionary/map from libtype (e.g. nlib, hlib, etc) to cross-section identifier (e.g. 80c, 71h, etc).  It could then be added to the end of the MCNP material string.

### PR DESCRIPTION
**Pull request by mehmeturkmen**
This PR resolves the issue #1165. For now, the code requires a separate line or separate lines of default library naming. All the default library types were added, including nlib, hlib, elib and so on. 
Output: 
{1: pyne.material.Material({80160000: 0.118619, 922340000: 0.0006705, 922350000: 0.0846125, 922380000: 0.796098}, -1.0, 10.4, -1.0, {"mat_number":"1","table_ids":{"80160":{"hlib":"66h","nlib":"12c","plib":"12p"},"922340":{"hlib":"66h","nlib":"12c","plib":"12p"},"922350":{"hlib":"66h","nlib":"12c","plib":"12p"},"922380":{"hlib":"66h","nlib":"12c","plib":"12p"}}})}
 
"tests/mcnp_inp.tex" was modified but not "tests/test_mcnp.py" as some confusing lines. So it may need some changes for passing the test.

### CODE REVIEW

**Comment by gonuke** [comment]
Thanks @mehmeturkmen - this is a valuable addition, but I suggest adding a test for this.  You have updated the test input, but haven't added a test that tests for this new behavior, as far as I can tell.

(Note that the issue you reference is for an MCNP writer and not an MCNP reader, but this is useful nevertheless.)

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: I wonder if you could just loop through the entries in `line.split()` to see if any of them match?  This would allow the library indicator to be at any position, and also allow there to be multiple library indicators per line.

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: If using my above plan, you'll need to set some flag rather than use this new test.

**Review by gonuke — CHANGES_REQUESTED** [review]
I had a thought on allowing the library indicator to come at any point in the line.

**Comment by gonuke** [comment]
I've updated the name of this PR so that we know what it is from the list. 

Also, best practice would be making a separate branch on your own fork for this rather than using the `0.7.0-rc` branch (even though it is on your own fork)

File: pyne/mcnp.py
Comment (mehmeturkmen) [reply pyne/mcnp.py]: This modification works for any combination of multiple library indicators (including all in a single line) as long as any library indicator begins in a separate line or lines. I also tried to extend this modification through a mixture of library indicator and isotope indicator along with a fraction indicator but I did not achieve yet. As I understood from the function of `mat_from_inp_line`, searching for all possible cases needs a deep change in the considered function due to the way how it was written. But still there may be a short-cut. I will be working on it ! Besides the reader ability, I am going to add the writer capability for default library in this PR.

File: CHANGELOG.rst
Comment (nsryan2) [inline CHANGELOG.rst]:
```suggestion
  * Default library reading capability from material card, such as nlib, plib and etc. (#1269)
```

Comment (mehmeturkmen) [reply CHANGELOG.rst]: Done!

**Comment by mehmeturkmen** [comment]
[x] Default library identifier reading capablity
[x] Test file was created for all possible combinations of library and atomic data
[x] Change log file conflict was resolved. 
[x] MCNP input file for testing was modified.

File: CHANGELOG.rst
Comment (gonuke) [inline CHANGELOG.rst]:
```suggestion
* File scripts/activation_responses.py to get activation responses data for ALARA output file for better visualization. Responses include:
```
This needs to be one line to get the formatting right

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: These IDs are attributes of the whole material, not each nuclide.  For example, if someone manipulated the material to have addition nuclides (e.g. depletion/transmutation) we'd want the new nuclides to inherit this info.

They can be added as separate metadata, or as a stand alone entry in `table_ids`.  
E.g: `table_ids['default'] = lib_id` at some location that is not nuclide specfic, like when `table_ids` is first declared, above

Also, it should only be used if it's not empty.

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: Remove blank line (PEP8?)

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: Remove blank line (PEP8?)

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: Remove blank line (PEP8?)

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: Why not do this at about 1443 where you are processing `data_string`?  This is a code maintainability question related to separation of concerns.  The overarching purpose of this loop is to filter out information that doesn't need to be processed.  The next loop will goes through this filtered string to process the necessary info.  It makes more sense to process the default lib info there.

**Review by gonuke — CHANGES_REQUESTED** [review]
Making good progress - I think this is all functional, but I have some style/maintainability and design requests.

File: pyne/mcnp.py
Comment (mehmeturkmen) [reply pyne/mcnp.py]: I thought that to assign a default library identifier to each isotope would be more useful in the future pyNE releases. But, as you said, since all isotopes in a material card would use the default libraries, if exists, we can specify the lib as a separate metadata. So, your point definitely makes sense. In this case, I will try to add `lib_id` in `table_ids` with `default` key name in order to be more specific.

File: pyne/mcnp.py
Comment (bam241) [inline pyne/mcnp.py]:
```suggestion
        # This line reads default cross-section library from material card 
```

File: pyne/mcnp.py
Comment (bam241) [inline pyne/mcnp.py]:
```suggestion
```

File: pyne/mcnp.py
Comment (bam241) [inline pyne/mcnp.py]:
```suggestion

```

File: pyne/mcnp.py
Comment (bam241) [inline pyne/mcnp.py]:
```suggestion

class Wwinp(Mesh):
```
I believe PEP8 requires 2 blank lines bewteen classes

**Review by bam241 — CHANGES_REQUESTED** [review]
This looks good.

Some few minor formatting comments.

I'll let @gonuke merge that one as he has unresolved comments

**Comment by mehmeturkmen** [comment]
@bam241 Do not merge yet. I am working on it. I am going to finalize it soon.

File: pyne/mcnp.py
Comment (mehmeturkmen) [reply pyne/mcnp.py]: Solved!

File: pyne/mcnp.py
Comment (mehmeturkmen) [reply pyne/mcnp.py]: Solved!

**Comment by mehmeturkmen** [comment]
I think this PR now is done! Seems ready to merge.

File: pyne/mcnp.py
Comment (gonuke) [inline pyne/mcnp.py]: I just realized that this won't work!!!  The logic of this `i & 1 == 1` threw me off

**Review by gonuke — CHANGES_REQUESTED** [review]

File: pyne/mcnp.py
Comment (mehmeturkmen) [reply pyne/mcnp.py]: Still there may be a way to get over it! I am on it.

**Comment by mehmeturkmen** [comment]
It seems we have a bunch of fails. I have rebased my branch so can we restart tests?

**Comment by bam241** [comment]
you need to rebase the branch so the test gets their latest version.

They are basically falling because they are out of date

**Comment by gonuke** [comment]
There is a new PR from me into your fork, @mehmeturkmen, that I think gets it all right.  It was a little more subtle than we expected.  I've also introduced some more variables to that it is more readable and maintainable in the long term.

I think you'll also need a rebase.

**Comment by gonuke** [comment]
I think this needs a rebase still to update for new masses

**Comment by gonuke** [comment]
This was absorbed in #1316 to finalize the expected results for the tests based on the updated mass data

**Pull request closed without merging**
